pkg/cloudprovider/vm.go

```diff
@@ -168,16 +169,39 @@ func (i *vmIdentityInfo) SetUserIdentities(ids map[string]bool) bool {
 
 	// all identities are the node are to be removed
 	if len(nodeList) == 0 {
-		i.info.UserAssignedIdentities = nil
-		if i.info.Type == compute.ResourceIdentityTypeSystemAssignedUserAssigned {
-			i.info.Type = compute.ResourceIdentityTypeSystemAssigned
-		} else {
-			i.info.Type = compute.ResourceIdentityTypeNone
-		}
+		i.ResetResourceIdentityType()
 		return true
 	}
 
 	i.info.Type = getUpdatedResourceIdentityType(i.info.Type)
 	i.info.UserAssignedIdentities = userAssignedIdentities
 	return len(i.info.UserAssignedIdentities) > 0
 }
+
+func (i *vmIdentityInfo) RemoveUserIdentity(delID string) bool {
+	removed := false
+	if i.info.UserAssignedIdentities != nil {
+		for id := range i.info.UserAssignedIdentities {
```

Comment:
`i.info.UserAssignedIdentities` is a map, so you can just check if the `delID` exists in the map instead of the for loop